packages/@aws-cdk/aws-lambda-event-sources/lib/stream.ts

```diff
@@ -21,6 +21,50 @@ export interface StreamEventSourceProps {
    */
   readonly batchSize?: number;
 
+  /**
+   * If the function returns an error, split the batch in two and retry.
+   *
+   * @default false
+   */
+  readonly bisectBatchOnError?: boolean;
+
+  /**
+   * An Amazon SQS queue or Amazon SNS topic destination for discarded records.
+   *
+   * @default discarded records are ignored
+   */
+  readonly onFailure?: lambda.IDestination;
```

Comment:
Still need to address the comment on not using `IDestination`.

Reply:
I think ideally this would just accept a raw SNS or SQS object and do the rest for the end user.  Not entirely sure how I would get that to work with typing.

This is what I'm thinking from a consumption stand point, but how might I reconcile that with SNS, SQS, and JSII

`
        const dlq = new sqs.Queue(this, 'Q');
        fn.addEventSource(new KinesisEventSource(stream, {
            startingPosition: lambda.StartingPosition.TRIM_HORIZON,
            onFailure: dlq
        }));
`